pandas/core/dtypes/base.py

```diff
@@ -118,12 +121,15 @@ def is_dtype(cls, dtype):
 
         1. ``cls.construct_from_string(dtype)`` is an instance
            of ``cls``.
-        2. 'dtype' is ``cls`` or a subclass of ``cls``.
+        2. ``dtype`` is an object and is an instance of ``cls``
+        3. 'dtype' is a class and is ``cls`` or a subclass of ``cls``.
         """
         if isinstance(dtype, str):
             try:
                 return isinstance(cls.construct_from_string(dtype), cls)
             except TypeError:
                 return False
-        else:
+        elif inspect.isclass(dtype):
```

Comment:
you have 2 conditions the same here

Reply:
`issubclass` vs. `isinstance` :) Want to be able to handle `.is_dtype(ExtensionDtype)` and `.is_dtype(ExtensionDtype())`.